tests/extractors/test_bigquery_extractor.py

```diff
@@ -103,35 +111,35 @@ def test_extract_on_complete(self, mock_json, mock_client, mock_ti):
             dag_id="dag_id",
         )
 
-        job_name = get_job_name(task=task)
-        BigQueryExtractor(task).extract_on_complete(mock_ti)
+        ti = TestBigQueryExtractor.get_ti(task=task)
+        steps_meta = BigQueryExtractor(task).extract_on_complete(ti)
 
-        mock_ti.return_value = TestBigQueryExtractor.get_ti(self)
-        log.info(mock_ti.return_value.job_id)
-        mock_client.get_job(job_name).return_value = {}
-        mock_json.return_value = \
-            TestBigQueryExtractor.get_job_details(self)
+        mock_xcom_pull.assert_called_once_with(
+            task_ids=ti.task_id, key='job_id')
+        mock_xcom_pull.return_value = "job_0ma4ph-Bn00JazZxVVsD7ybzWN3E"
 
-        assert mock_json.return_value.get("statistics").get(
-            "totalBytesProcessed") == "65935918"
+        mock_get_job.assert_called_once()
 
-    @staticmethod
-    def get_job_details(self):
-        with open('tests/extractors/job_details.json') as json_file:
-            return json.load(json_file)
+        job_details_str = file_to_str("tests/extractors/job_details.json")
+        job_details_obj = str_to_obj(job_details_str)
+        mock_get_job.return_value = job_details_obj
 
-    @staticmethod
-    def get_ti(self):
-        # TODO: TaskInstance represents a row in table task_instance
-        # Update this logic
-        # return TaskInstance
-        class TaskInstance:
-            job_id = None
+        mock_json.assert_called_once()
+        mock_json.return_value = jsonpickle.decode(job_details_str)
 
-            def __init__(self, job_id):
-                self.job_id = job_id
+        assert mock_json.return_value._properties['jobReference']['jobId']\
+               == mock_xcom_pull.return_value
```

Comment:
why are you asserting that your mock return values are equal? you want to check that the actual return values is matching what you expect. Here you are comparing you own test setup data to itself?

Reply:
done.